pkg/supportbundle/supportbundle.go

```diff
@@ -237,9 +237,77 @@ func CreateSupportBundleDependencies(appID string, sequence int64, opts types.Tr
 	return &supportBundleObj, nil
 }
 
-// CreateSupportBundleAnalysis takes a support bundle, unpacks to analyze then adds the analysis to
-// the support bundle secret
+func getAnalysisFromBundle(archivePath string) ([]byte, error) {
+	bundleDir, err := ioutil.TempDir("", "kots")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create temp dir")
+	}
+	defer os.RemoveAll(bundleDir)
+
+	tarGz := archiver.TarGz{
+		Tar: &archiver.Tar{
+			ImplicitTopLevelFolder: false,
+		},
+	}
+	if err := tarGz.Unarchive(archivePath, bundleDir); err != nil {
+		return nil, errors.Wrap(err, "failed to unarchive")
+	}
+
+	var analysis []byte
+	err = filepath.Walk(bundleDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		relPath, err := filepath.Rel(bundleDir, path)
+		if err != nil {
+			return errors.Wrap(err, "failed to get relative path")
+		}
+
+		// trim the subdirectory from the path too (if any)
+		trimmedRelPath := SupportBundleNameRegex.ReplaceAllString(relPath, "")
+		trimmedRelPath = strings.TrimPrefix(trimmedRelPath, string(os.PathSeparator))
+		if trimmedRelPath == "" {
+			return nil
+		}
+
+		if trimmedRelPath == "analysis.json" {
```

Comment:
We should just hardcode absolute path to avoid returning some random file with the same name since there is no schema validation.